pkg/index/job/save/service/indexer_test.go

```diff
@@ -0,0 +1,417 @@
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			addrs := []string{
+				"127.0.0.1:8080",
+			}
+			return test{
+				name: "Success: when there is no error in the save indexing request process",
+				args: args{
+					ctx: context.Background(),
+				},
+
+				fields: fields{
+					client: &mockDiscovererClient{
+						GetAddrsFunc: func(_ context.Context) []string {
+							return addrs
+						},
+						GetClientFunc: func() grpc.Client {
+							return &mockGrpcClient{
+								OrderedRangeConcurrentFunc: func(_ context.Context, _ []string, _ int,
+									_ func(_ context.Context, _ string, _ *grpc.ClientConn, _ ...grpc.CallOption) error,
+								) error {
+									return nil
+								},
+							}
+						},
+					},
+				},
+			}
+		}(),
+		func() test {
+			addrs := []string{
+				"127.0.0.1:8080",
+			}
+			return test{
+				name: "Fail: when there is an error wrapped with gRPC status in the save indexing request process",
+				args: args{
+					ctx: context.Background(),
+				},
+				fields: fields{
+					client: &mockDiscovererClient{
+						GetAddrsFunc: func(_ context.Context) []string {
+							return addrs
+						},
+						GetClientFunc: func() grpc.Client {
+							return &mockGrpcClient{
+								OrderedRangeConcurrentFunc: func(_ context.Context, _ []string, _ int,
+									_ func(_ context.Context, _ string, _ *grpc.ClientConn, _ ...grpc.CallOption) error,
+								) error {
+									return status.WrapWithInternal(
+										agent.SaveIndexRPCName+" API connection not found",
+										errors.ErrGRPCClientConnNotFound("*"),
+									)
+								},
+							}
+						},
+					},
+				},
+				want: want{
+					err: status.Error(codes.Internal,
+						agent.SaveIndexRPCName+" API connection not found"),
+				},
+			}
+		}(),
+		func() test {
+			addrs := []string{
+				"127.0.0.1:8080",
+			}
+			return test{
+				name: "Fail: When the OrderedRangeConcurrent method returns a gRPC client conn not found error",
+				args: args{
+					ctx: context.Background(),
+				},
+
+				fields: fields{
+					client: &mockDiscovererClient{
+						GetAddrsFunc: func(_ context.Context) []string {
+							return addrs
+						},
+						GetClientFunc: func() grpc.Client {
+							return &mockGrpcClient{
+								OrderedRangeConcurrentFunc: func(_ context.Context, _ []string, _ int,
+									_ func(_ context.Context, _ string, _ *grpc.ClientConn, _ ...grpc.CallOption) error,
+								) error {
+									return errors.ErrGRPCClientConnNotFound("*")
+								},
+							}
+						},
+					},
+				},
+				want: want{
+					err: status.Error(codes.Internal,
+						agent.SaveIndexRPCName+" API connection not found"),
+				},
+			}
+		}(),
+		func() test {
+			targetAddrs := []string{
+				"127.0.0.1:8080",
+			}
+			targetAddrList := map[string]bool{
+				targetAddrs[0]: true,
+			}
```

Comment:
It seems that targetAddr is registered in targetAddrList here, so why would the test fail?

Reply:
Sorry for the confusion 🙏 
This test is meant to be a test case for when the address the user expects does not actually exist. 
So GetAddrs returns nil. 👉 https://github.com/vdaas/vald/blob/feature/index-job/save/pkg/index/job/save/service/indexer_test.go#L167]

I'll add a comment as well!

Reply:
I have fixed! 👉 https://github.com/vdaas/vald/pull/2227/commits/fc44305036bdd13ee29d2c676d73e490ed7cefa3